tests/components/elmax/test_config_flow.py

```diff
@@ -26,97 +26,71 @@
 CONF_POLLING = "polling"
 
 
-def _has_error(errors):
-    return errors is not None and len(errors.keys()) > 0
-
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
 
-async def _bootstrap(
-    hass,
-    source=config_entries.SOURCE_USER,
-    username=MOCK_USERNAME,
-    password=MOCK_PASSWORD,
-    panel_name=MOCK_PANEL_NAME,
-    panel_pin=MOCK_PANEL_PIN,
-) -> FlowResult:
 
-    result = await hass.config_entries.flow.async_init(
-        DOMAIN, context={"source": source}
+async def test_standard_setup(hass):
+    """Test the standard setup case."""
+    # Setup once.
+    show_form_result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
-    if result["type"] != data_entry_flow.RESULT_TYPE_FORM or _has_error(
-        result["errors"]
-    ):
-        return result
-    result2 = await hass.config_entries.flow.async_configure(
-        result["flow_id"],
+    login_result = await hass.config_entries.flow.async_configure(
+        show_form_result["flow_id"],
         {
-            CONF_ELMAX_USERNAME: username,
-            CONF_ELMAX_PASSWORD: password,
+            CONF_ELMAX_USERNAME: MOCK_USERNAME,
+            CONF_ELMAX_PASSWORD: MOCK_PASSWORD,
         },
     )
-    if result2["type"] != data_entry_flow.RESULT_TYPE_FORM or _has_error(
-        result2["errors"]
-    ):
-        return result2
-    result3 = await hass.config_entries.flow.async_configure(
-        result2["flow_id"],
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We should patch the integration setup function on all create entry and successful reauth results to avoid setting up the whole integration. See the scaffold script test template for example.

Reply:
Done!